tools/2024-06-gitops-feature-spec.md

```diff
@@ -0,0 +1,362 @@
+
+## Desired customer experience outcome
+<!-- <Write this as an “I statement” that expresses the new capability from customer perspective … i.e. After this scenario is implemented “I can do, x, y, z, steps in cloud native app developer and seamlessly blah blah blah …. As a result <summarize positive impact on your work / business>  -->
+With the implementation of this scenario, I can deploy and manage Radius Environments, their associated Recipes, Applications, and Resources via my existing GitOps (Flux and ArgoCD) toolsets. This means that I don't have to author pipelines or manually run Radius commands to deploy Radius-managed resources. Instead, I can leverage a git repo as the source of truth for my Radius-managed resources and use my existing GitOps tooling to deploy and manage those resources. This ensures that my experience with managing Radius resources is consistent with how I normally use GitOps to manage Kubernetes clusters and applications.
+
+> There is no new Radius specific user experience for this scenario. Customers are only interacting with Flux or ArgoCD. However, note that we're combining two different but existing user experiences (i.e. GitOps `git push` + Radius `app.bicep`).
+
+### Detailed Customer Experience
+ <!-- <List of steps the customer goes through from the start to the end of the scenario to provide more detailed view of exactly what the customer is able to do given the new capabilities>  -->
+<!-- Step 1
+Step 2
+… -->
+
+**Step 1: Define and deploy Radius Environment and Recipes**
+
+As an operator, I need to define and deploy a Radius Environment and associate relevant Recipes, Credentials, etc. to that environment so that my development team can build and deploy their Radius applications.
+
+1. Using my editor of choice, such as VSCode, I open an existing Radius environment file, `env.bicep`, from the online Radius samples repo. 
+1. I use Radius documentation to determine specific edits to my `env.bicep` file such as Kubernetes cluster and the appropriate cloud provider registrations and credentials. 
+   1. I will create a Radius Environment resource and add all the information listed above. 
+   1. I will then edit my Radius Environment resource to contain the properties required to register my Radius Recipe such as template path, template kind, etc.
+1. I then push my new `env.bicep` file to my git repository, as I would any other file. 
+1. This triggers Flux or ArgoCD to detect the change, read the new `env.bicep` file, and deploy the Radius Environment as specified.
```

Comment:
nit: it's not really flux or argocd that deploys the Radius Environment, flux or argocd tell Radius to deploy it.